util/util.go

```diff
@@ -0,0 +1,291 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+
+	"github.com/lbryio/lbry.go/extras/errors"
+	"github.com/lbryio/lbry.go/lbrycrd"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/mitchellh/go-ps"
+	log "github.com/sirupsen/logrus"
+)
+
+func GetBlobsDir() string {
+	blobsDir := os.Getenv("BLOBS_DIRECTORY")
+	if blobsDir == "" {
+		usr, err := user.Current()
+		if err != nil {
+			log.Error(err.Error())
+			return ""
+		}
+		blobsDir = usr.HomeDir + "/.lbrynet/blobfiles/"
+	}
+
+	return blobsDir
+}
+
+func IsBlobReflectionOff() bool {
+	return os.Getenv("REFLECT_BLOBS") == "false"
+}
+
+func GetLBRYNetDir() string {
+	lbrynetDir := os.Getenv("LBRYNET_DIR")
+	if lbrynetDir == "" {
+		usr, err := user.Current()
+		if err != nil {
+			log.Errorln(err.Error())
+			return ""
+		}
+		return usr.HomeDir + "/.lbrynet/"
+	}
+	return lbrynetDir
+}
+
+const ALL = true
+const ONLINE = false
+
+func GetLBRYNetContainer(all bool) (*types.Container, error) {
+	return getDockerContainer("lbrynet", all)
+}
+
+func getDockerContainer(name string, all bool) (*types.Container, error) {
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		panic(err)
+	}
+	filters := filters.NewArgs()
+	filters.Add("name", name)
+	containers, err := cli.ContainerList(context.Background(), types.ContainerListOptions{All: all, Filters: filters})
+	if err != nil {
+		panic(err)
+	}
+	for _, container := range containers {
+		fmt.Printf("%s %s\n", container.ID[:10], container.Image)
+	}
+	if len(containers) == 0 {
+		return nil, nil
+	}
+	if len(containers) > 1 {
+		return nil, errors.Err("more than one %s container found", name)
+	}
+
+	return &containers[0], nil
+}
+
+func IsUsingDocker() bool {
+	return os.Getenv("LBRYNET_USE_DOCKER") == "true"
+}
+
+func IsRegTest() bool {
+	return os.Getenv("REGTEST") == "true"
```

Comment:
what if it's TRUE or TrUe or 1?
should we not just assume that if the env var is set then it's regtest?

Reply:
I thought about that and its not recommended because then someone could set it to false and expect certain things. Since this is so important, I thought it best to be explicit.

Reply:
i improved this today